### PR TITLE
Add press-packages

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

**Comment by nlisgo** [comment]
@thewilkybarkid I thought I would start this off with a pull request which borrows heavily from the blog articles. It is my understanding that we can reference at least one article, in fact we may require at least one article. Also, am I right in thinking that the subjects should be gathered from the referenced articles rather than input on the press package level?

**Comment by thewilkybarkid** [comment]
Looking good. Almost certain that they have to reference at least 1 article. And the subjects of the press package are computed from the articles (not sure about ordering, maybe article 1 subjects + article 2 subjects + ... then make them unique?), but for the API definition that doesn't matter (the implementation can always be changed to be curated list without the schema changing).

**Comment by nlisgo** [comment]
I've added support for relatedContent. The media contacts could use the `misc/person.v1.yaml` but add the email and telephone property?

File: src/model/press-package.v1.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                oneOf:
+                  - $ref: ../snippets/article-poa.v1.yaml
+                  - $ref: ../snippets/article-vor.v1.yaml
+            minItems: 1
+            uniqueItems: true
+        mediaContacts:
+            title: Media contacts
+            type: array
+            items:
+                $ref: ../misc/person.v1.yaml
```
Comment (nlisgo) [inline src/model/press-package.v1.yaml]: @thewilkybarkid We need to allow an array of people with the option properties of affiliation, email addresses and phone numbers. Please can you advise on the best approach for this.

Comment (thewilkybarkid) [reply src/model/press-package.v1.yaml]: We should do something like `author.v1.yaml` and add in the extra properties (so like https://github.com/elifesciences/api-raml/blob/develop/src/misc/author.v1.yaml#L31-L35 and https://github.com/elifesciences/api-raml/blob/develop/src/misc/author.v1.yaml#L46-L49). So here it'd be an `allOf` with the first item being the `$ref` and the second item being the two new properties.

Comment (thewilkybarkid) [reply src/model/press-package.v1.yaml]: So in the style of https://github.com/elifesciences/api-raml/blob/develop/src/misc/author.v1.yaml#L55-L61

Comment (nlisgo) [reply src/model/press-package.v1.yaml]: Cool. Will give this a go.

File: src/model/press-package.v1.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                  - $ref: ../snippets/article-vor.v1.yaml
+            minItems: 1
+            uniqueItems: true
+        mediaContacts:
+            title: Media contacts
+            type: array
+            items:
+                allOf:
+                  - $ref: ../misc/person.v1.yaml
+                  - properties:
```
Comment (nlisgo) [inline src/model/press-package.v1.yaml:28]: @thewilkybarkid how does this look? I will extend the sample fixtures once you are ok with this.

File: src/model/press-package.v1.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                  phoneNumbers:
+                      type: array
+                      items:
+                          $ref: phone-number.v1.yaml
+                  postalAddresses:
+                      type: array
+                      items:
+                          $ref: address.v1.yaml
+    required:
+      - content
```
Comment (thewilkybarkid) [inline src/model/press-package.v1.yaml:43]: I suggest we make `relatedContent` required. If we later find out it's optional we can always change it.

File: src/model/press-package.v1.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            items:
+                allOf:
+                  - $ref: ../misc/person.v1.yaml
+                  - properties:
+                      emailAddresses:
+                          type: array
+                          items:
+                              type: string
+                              format: email
+                  phoneNumbers:
```
Comment (thewilkybarkid) [inline src/model/press-package.v1.yaml]: Indentation needs fixing.

File: src/model/press-package.v1.yaml (hunk 2 above)
Comment (thewilkybarkid) [reply src/model/press-package.v1.yaml:28]: 👍

**Comment by nlisgo** [comment]
@thewilkybarkid I believe I have addressed your latest feedback.

**Comment by nlisgo** [comment]
Just going to rebase

File: src/snippets/press-package.v1.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: Press package snippet
+type: object
+properties:
+    id:
+        description: ID
+        type: string
+        minLength: 1
```
Comment (thewilkybarkid) [inline src/snippets/press-package.v1.yaml]: I've just merged #50, could you change this to a `$ref: ../misc/id.v1.yaml`?

Comment (nlisgo) [reply src/snippets/press-package.v1.yaml]: Sure

**Comment by nlisgo** [comment]
@thewilkybarkid I have extended the sample json fixture. I believe that affiliations is more appropriate than postalAddresses. Please review.

**Pull request merged**